src/contract/lido/lido-stETH-contract.ts

```diff
@@ -0,0 +1,28 @@
+import { Contract, ContractRunner } from "ethers";
+
+import { LidoSTETH } from "typechain";
+import lidoSTETHAbi from "../../abi/lido/LidoSTETH.json"
+
+export class LidoSTETHContract {
+
+    private readonly contract: LidoSTETH;
+    constructor(contractAddress: string, provider?: ContractRunner) {
+        this.contract = new Contract(contractAddress, lidoSTETHAbi, provider) as unknown as LidoSTETH;
+    }
+
+    async submit(amount: bigint, referralAddress: string) {
+        const tx = await this.contract.submit.populateTransaction(referralAddress);
+        tx.value = amount;
+        return tx;
+    }
+
+    async balanceOf(address: string): Promise<bigint> {
+        const result = await this.contract.balanceOf(address);
+        return result;
+    }
+
+    async sharesOf(address: string): Promise<bigint> {
+        const result = await this.contract.sharesOf(address);
+        return result;
+    }
```

Comment:
If I'm not wrong I think you could do this change, since thse functions are still returning a promise.
```suggestion
    async balanceOf(address: string): Promise<bigint> {
        return this.contract.balanceOf(address);
    }

    async sharesOf(address: string): Promise<bigint> {
        return this.contract.sharesOf(address);
    }
```

Reply:
`async` could even be stripped here, as there is no async logic going on anymore. The result of the functions would of course still have to be `await`ed to unwrap

Reply:
Agree